packages/devtools_app/lib/src/screens/logging/logging_screen_v2/logging_table_v2.dart

```diff
@@ -0,0 +1,369 @@
+                  border: OutlineInputBorder(),
+                  hintText: 'Filter',
+                ),
+                onSubmitted: (value) {},
+              ),
+            ),
+            const Expanded(
+              child: TextField(
+                decoration: InputDecoration(
+                  border: OutlineInputBorder(),
+                  hintText: 'Search',
+                ),
+              ),
+            ),
+          ],
+        ),
+        Expanded(
+          child: _LoggingTableContextMenu(
+            child: _LoggingTableContents(
+              model: widget.model,
+              verticalController: _verticalController,
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _LoggingTableContents extends StatefulWidget {
+  _LoggingTableContents({
+    required this.model,
+    required ScrollController verticalController,
+  }) : _verticalController = verticalController;
+
+  static const _millisecondsUntilCacheProgressShows = 500;
+  static const _millisecondsUntilCacheProgressHelperShows = 2000;
+
+  final LoggingTableModel model;
+  final ScrollController _verticalController;
+
+  @override
+  State<_LoggingTableContents> createState() => _LoggingTableContentsState();
+}
+
+class _LoggingTableContentsState extends State<_LoggingTableContents> {
+  final Stopwatch _progressStopwatch = Stopwatch();
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (context, constraints) {
+        widget.model.tableWidth = constraints.maxWidth;
+        _progressStopwatch.reset();
+        return ListenableBuilder(
+          listenable: widget.model.cacheLoadProgress,
+          builder: (context, _) {
+            final cacheLoadProgress = widget.model.cacheLoadProgress.value;
+            if (cacheLoadProgress != null) {
+              double progress = cacheLoadProgress;
+              if (!_progressStopwatch.isRunning) {
+                _progressStopwatch.start();
+              }
+
+              if (_progressStopwatch.elapsedMilliseconds <
+                  _LoggingTableContents._millisecondsUntilCacheProgressShows) {
+                progress = 0.0;
+              }
+
+              return Center(
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    const Text(
+                      'Resizing... this will only take a moment.',
+                    ),
+                    const SizedBox(height: defaultSpacing),
+                    SizedBox(
+                      width: defaultLinearProgressIndicatorWidth,
+                      height: defaultLinearProgressIndicatorHeight,
+                      child: LinearProgressIndicator(
+                        value: progress,
+                      ),
+                    ),
+                    const SizedBox(height: defaultSpacing),
+                    if (_progressStopwatch.elapsedMilliseconds >
```

Comment:
this build method is not retriggered for updated to the stopwatch elapsed time is it?

Reply:
I'm not sure I understand the question. Could you please elaborate?

Reply:
From what I can tell, we are not rebuilding this widget when the value of `_progressStopwatch.elapsedMilliseconds` changes. So how will this if statement be hit? Is this build method triggered again later on once we expect the stopwatch to have progressed?

Reply:
Progress updates will refresh every chunk so this should get hit enough to be fresh.
Future changes can be made if people like to have huge retention limits